doc/devguide/codebase/unittests-c.rst

```diff
@@ -1,12 +1,152 @@
+
+.. code-block:: c
+
+    void MyUnitTest(void)
+    {
+        int n = 1;
+        void *p = NULL;
+    
+        FAIL_IF(n != 1);
+        FAIL_IF_NOT(n == 1);
+        FAIL_IF_NOT_NULL(p);
+        FAIL_IF_NULL(p);
+    
+        PASS;
+    }
+
+Each unit test needs to be registered with ``UtRegisterTest()``. Example::
+
+    UtRegisterTest("MyUnitTest", MyUnitTest);
+
+where the first argument is the name of the test, and the second argument is the function. Existing modules should already have a function that registers its unit tests. Otherwise the unit tests will need to be registered. Look for a module similar to your new module to see how best to register the unit tests or ask the development team for help.
+
+Examples
+--------
+
+From ``conf-yaml-loader.c``:
+
+.. code-block:: c
+
+    /**
+     * Test that a configuration section is overridden but subsequent
+     * occurrences.
+     */
+    static int
+    ConfYamlOverrideTest(void)
+    {
+        char config[] =
+            "%YAML 1.1\n"
+            "---\n"
+            "some-log-dir: /var/log\n"
+            "some-log-dir: /tmp\n"
+            "\n"
+            "parent:\n"
+            "  child0:\n"
+            "    key: value\n"
+            "parent:\n"
+            "  child1:\n"
+            "    key: value\n"
+            ;
+        const char *value;
+    
+        ConfCreateContextBackup();
+        ConfInit();
+    
+        FAIL_IF(ConfYamlLoadString(config, strlen(config)) != 0);
+        FAIL_IF_NOT(ConfGet("some-log-dir", &value));
+        FAIL_IF(strcmp(value, "/tmp") != 0);
+    
+        /* Test that parent.child0 does not exist, but child1 does. */
+        FAIL_IF_NOT_NULL(ConfGetNode("parent.child0"));
+        FAIL_IF_NOT(ConfGet("parent.child1.key", &value));
+        FAIL_IF(strcmp(value, "value") != 0);
+    
+        ConfDeInit();
+        ConfRestoreContextBackup();
+    
+        PASS;
+    }
+
+In ``detect-ike-chosen-sa.c``, it is possible to see the freeing of resources (``DetectIkeChosenSaFree``) and the
+function that should group all the ``UtRegisterTest`` calls:
+
+.. code-block:: c
+
+    #ifdef UNITTESTS
+    .
+    .
+    .
+    static int IKEChosenSaParserTest(void)
+    {
+        DetectIkeChosenSaData *de = NULL;
+        de = DetectIkeChosenSaParse("alg_hash=2");
+    
+        FAIL_IF_NULL(de);
+        FAIL_IF(de->sa_value != 2);
+        FAIL_IF(strcmp(de->sa_type, "alg_hash") != 0);
+    
+        DetectIkeChosenSaFree(NULL, de);
+        PASS;
+    }
+    
+    #endif /* UNITTESTS */
+    
+    void IKEChosenSaRegisterTests(void)
+    {
+    #ifdef UNITTESTS
+        UtRegisterTest("IKEChosenSaParserTest", IKEChosenSaParserTest);
+    #endif /* UNITTESTS */
+
+Unit Tests - Rust codebase
```

Comment:
Do we need this here since we have a separate page for rust tests?

Reply:
Victor said it could be good to mention the existence of rust unittests, too, and direct them to the proper page. Maybe I can think of a better way to doing so...

Reply:
Perhaps, we can mention both on the index page for testing?

Reply:
Would make sense, I think, considering how it is with the changes :) I will try it that way!